### DISCUSSION TITLE
Delay of authelia service registration in traefik

### DISCUSSION THREAD

**Discussion by asherikov** [authelia/authelia#6823] [Documentation Feedback]
Traefik registers authelia service only after receiving a health check docker event, so there seem to be a random, up to 30 sec delay in bringing it up (https://github.com/authelia/authelia/blame/master/Dockerfile#L32). Until this happens an attempt to access authentication service results in 404 error, which might be related to https://github.com/authelia/authelia/issues/2476. It took me quite some time to figure this out, so it might be worth mentioning in https://www.authelia.com/integration/proxies/traefik/, e.g., one option could be adding this to traefik service in compose file:
```
depends_on:
    authelia:
        condition: service_healthy
```

**Comment by glime-ai** [comment]
Please note this response was created using generative AI leveraging previously available information from the repository. This information may be helpful however it is our strong recommendation that you carefully consider the information before blindly applying it.

---

Based on the context provided, it seems you're encountering an issue where Traefik only registers the Authelia service after receiving a Docker health check event, leading to a delay of up to 30 seconds before the Authelia service becomes available. This delay can result in a 404 error when attempting to access the authentication service, as discussed in https://github.com/authelia/authelia/issues/2476.

Your suggestion to mitigate this issue involves using the `depends_on` condition `service_healthy` in the Docker Compose file for the Traefik service, ensuring that Traefik depends on Authelia being in a healthy state before starting. This approach is indeed a practical solution to ensure that dependent services are fully operational before others start, reducing the likelihood of encountering 404 errors due to service unavailability.

However, it's important to note that the `depends_on` option with `condition: service_healthy` is only available in version 2.1 to 2.4 of the Docker Compose file format and is not supported in version 3.x and above. For users utilizing version 3.x of the Docker Compose file format, an alternative approach would be required, such as implementing a custom script to check the health of the Authelia service before starting the Traefik service.

Given the potential impact of this issue on users integrating Authelia with Traefik, it would indeed be beneficial for this information and suggested workaround to be mentioned in the Authelia documentation for Traefik integration. I recommend reaching out to the Authelia team through their documentation repository or discussion forums to suggest adding this information to help future users encountering similar issues.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=6fsHkcyufjQLaB&utm_medium=github).

**Comment by asherikov** [comment]
The bot appears to be wrong, latest compose spec allows specifying conditions https://docs.docker.com/compose/compose-file/05-services/#depends_on

**Comment by hendrik1120** [comment]
Not sure if I understand your issue correctly, but this is not specific to authelia. Traefik won't forward any request to an unhealthy container. The configuration is just an example, and you should be familiar with the reverse proxy you are using. If you explicitly want to disable this behavior in traefik, take a look at this https://doc.traefik.io/traefik/providers/docker/#provider-configuration. There are also several ways to disable the configured health check. 

Traefik can also forward other services that don't require authelia, so depending traefik on authelia seems like a more specific use case to me. Again, it's just an example, so feel free to customize it to your specific needs.

**Comment by nightah** [comment]
Our interval for the health check is needlessly too high.

For now you can override it with the following in your compose definition:

```yaml
healthcheck:                                          
  interval: 4s
```

We will adjust the interval to probably be 3-4s.

**Comment by asherikov** [comment]
Ok, I see your point.
